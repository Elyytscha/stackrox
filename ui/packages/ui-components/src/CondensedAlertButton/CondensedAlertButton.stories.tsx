import React from 'react';
import { Meta, Story } from '@storybook/react/types-6-0';

import CondensedAlertButton from './CondensedAlertButton';

export default {
    title: 'CondensedAlertButton',
    component: CondensedAlertButton,
} as Meta;

export const DefaultButton: Story = () => {
    function onClick(): void {}
    return (
        <CondensedAlertButton type="button" onClick={onClick}>
            Click me
        </CondensedAlertButton>
    );
};
