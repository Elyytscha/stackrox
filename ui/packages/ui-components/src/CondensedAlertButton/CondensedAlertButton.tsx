import React, { ReactElement } from 'react';

import Button, { HOCButtonProps } from '../Button';

export type CondensedAlertButtonProps = HOCButtonProps;

function CondensedAlertButton({
    type,
    onClick,
    children,
}: CondensedAlertButtonProps): ReactElement {
    return (
        <Button type={type} onClick={onClick} colorType="alert" isCondensed>
            {children}
        </Button>
    );
}

export default CondensedAlertButton;
