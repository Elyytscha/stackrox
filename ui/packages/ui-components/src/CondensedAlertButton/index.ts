export { default } from './CondensedAlertButton';
export * from './CondensedAlertButton';
