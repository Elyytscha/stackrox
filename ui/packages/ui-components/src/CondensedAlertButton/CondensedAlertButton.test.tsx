import React from 'react';
import { render } from '@testing-library/react';

import CondensedAlertButton from './CondensedAlertButton';

describe('CondensedAlertButton', () => {
    test('renders the button', () => {
        function onClick(): void {}

        const { container, getByText } = render(
            <CondensedAlertButton type="button" onClick={onClick}>
                Click me!
            </CondensedAlertButton>
        );

        expect(getByText('Click me!')).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
});
