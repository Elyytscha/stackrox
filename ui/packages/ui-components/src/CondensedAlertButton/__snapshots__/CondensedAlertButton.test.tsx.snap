// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CondensedAlertButton renders the button 1`] = `
<div>
  <button
    class="border-2 font-600 inline-flex items-center justify-center rounded-sm border-alert-400 bg-alert-100 hover:bg-alert-200 hover:text-alert-700 text-alert-800 px-2"
    type="button"
  >
    Click me!
  </button>
</div>
`;
